crates/aggchain-proof-core/src/error.rs

```diff
@@ -1,5 +1,11 @@
 use serde::{Deserialize, Serialize};
 
+use crate::bridge::BridgeConstraintsError;
+
 /// Represents all the aggchain proof errors.
 #[derive(Clone, thiserror::Error, Debug, Serialize, Deserialize, PartialEq, Eq)]
-pub enum ProofError {}
+pub enum ProofError {
+    /// Error on the bridge constraints.
+    #[error(transparent)]
+    BridgeConstraintsError(#[from] BridgeConstraintsError),
```

Comment:
```suggestion
    BridgeConstraintsError(#[source] BridgeConstraintsError),
```

Reply:
I was initially with `#[source]` and without `#[error(transparent)]`, but I switched after https://github.com/agglayer/provers/pull/57#discussion_r1973808602
I am fine with either tbh, but it can't be `source` and `transparent` cc @Ekleog-Polygon

Reply:
We should stick with the error context for particular function/operation and `#[source]` to bubble up errors.